static/js/LayoutButtons.jsx

```diff
@@ -0,0 +1,56 @@
+import {useContext} from "react";
+import {ReaderPanelContext} from "./context";
+import {layoutOptions} from "./constants";
+import {InterfaceText} from "./Misc";
+
+const calculateLayoutState = () => {
+    const {language, textsData, panelMode} = useContext(ReaderPanelContext);
+    const primaryDir = textsData?.primaryDirection;
+    const translationDir = textsData?.translationDirection;
+    return (language !== 'bilingual') ? 'mono' //one text
+        : (primaryDir !== translationDir || panelMode === 'Sheet') ? 'mixed' //two texts with different directions
+            : (primaryDir === 'rtl') ? 'bi-rtl' //two rtl texts
+                : 'bi-ltr'; //two ltr texts
+}
+
+const LayoutButtons = () => {
+    const {language, textsData, setOption, layout, panelMode} = useContext(ReaderPanelContext);
+    const layoutState = calculateLayoutState();
+
+    const getPath = (layoutOption) => {
```

Comment:
current: use a single patterns for functions, either in the closure or outside of it. @edamboritz do you have a recommendation of a pattern here?

Reply:
ideally have distributed functions and call them into the functional component